app/models/user.rb

```diff
@@ -9,17 +10,19 @@ class User < ApplicationRecord
     uniqueness: {case_sensitive: false}
   validates :password, presence: true, allow_nil: true,
     length: {minimum: Settings.user_setting.password_minimum_length}
-  before_save :downcase_email
   has_secure_password
+  before_create :create_activation_digest
+  before_save :downcase_email
```

Comment:
before_save hay before_create chạy trước hả em?

Reply:
https://guides.rubyonrails.org/active_record_callbacks.html
"Here is a list with all the available Active Record callbacks, listed in the same order in which they will get called during the respective operations:"
before_save chay truoc a.

Reply:
vậy thì đặt `before_save :downcase_email` trước `before_create :create_activation_digest` nhé

Reply:
vang ah